server/api/controllers/image-controller.ts

```diff
@@ -44,42 +43,56 @@ export class ImageController {
   /*
   Handles the /api/image/upload endpoint 
   - handles all possible errors with the file upload (no file, no fileype, not recognized filetype)
-  - if validation passes, adds the request to the ImageRequestQueue
+  - if validation passes, adds the request with all uploaded files to the ImageRequestQueue
   */
-  static uploadImage(req: Request, res: Response) {
-    try {
-      const image = req.files?.image as UploadedFile;
 
-      if (!image) {
-        throw new Error("No image was uploaded");
+  static uploadImages(req: Request, res: Response) {
+    try {
+      // this is done because express-upload has no way to tell you whether one file or multiple files were uploaded
+      // casting singular file to an array, or keeping multiple files as an existing array
+      const images = (req.files?.images as UploadedFile[]).length
+        ? (req.files?.images as UploadedFile[])
+        : [req.files?.images as UploadedFile];
+
+      if (!images.length) {
+        throw new Error("No images were uploaded");
       }
 
-      if (!image.mimetype) {
-        throw new Error("No mimetype/filetype provided with the image.");
-      }
+      const errors: string[] = [];
+      const imagesWithFile: ImageWithFile[] = [];
 
-      if (
-        image.mimetype !== ImageMimetypes.JPEG &&
-        image.mimetype !== ImageMimetypes.PNG
-      ) {
-        throw new Error("Unsupported filetype uploaded.");
-      }
-      const transformedImage: Image = {
-        fileName: image.name,
-        fileType: image.name.split(".")[1],
-        path: this.getImagePath(image.name),
-        publicLink: this.getPublicLink(image.name),
-      };
+      for (const image of images) {
+        if (!image.mimetype) {
+          errors.push(`No mimetype/filetype provided with file ${image.name}`);
+          continue;
+        }
 
-      if (!this.queues[transformedImage.fileName]) {
-        this.queues[transformedImage.fileName] = new ImageRequestQueue();
+        if (
+          image.mimetype !== ImageMimetypes.JPEG &&
+          image.mimetype !== ImageMimetypes.PNG
+        ) {
+          errors.push(`Unsupported filetype for ${image.name}`);
+          continue;
+        }
+
+        const transformedImage: ImageWithFile = {
+          fileName: image.name,
+          fileType: image.name.split(".")[1],
```

Comment:
This could have weird behaviour with combination file types like `.jpeg.webp` - could we change it to combine all the elements of the split array beyond the 0th?
```suggestion
          fileType: image.name.split(".").slice(1).join('.'),
```

Reply:
i don't think this would happen since the fileUploader parses extension names safely so that all uploaded files have at most one extension (`server.ts`, line 36). updated regardless. In the case that someone uploads a file with combination filetypes, the file uploader concats the first extension into the file name, so `img.png.webp` would become `imgpng.webp`. I think this is desired behavior